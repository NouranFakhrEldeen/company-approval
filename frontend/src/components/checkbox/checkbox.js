/* eslint-disable no-unused-vars */

import React, { useState } from 'react';
import { capitalizeFirstWord } from '../../helpers';
import './checkbox.scss';


function checkbox({
  containerClasses,
  testID,
  id,
  name,
  spanValue,
  onChange,
  checked,
  //   checkbox,
  //   handleChange,
}) {
  const [checkbox, setCheckbox] = useState(true);

  const handleChange = (e) => {
    let value = e.target.value;
    if (e.target.type === 'checkbox') {
      //   value = e.target.checked;
      setCheckbox(!checkbox);
    }

  };
  return (
    <div className={containerClasses}>
      <label>
        <input
          type="checkbox"
          id={id}
          checked={checked}
          name={name}
          data-testid={'checkbox'}
          // data-testid={testID}
          // onChange={(e) => { handleChange(e); }}
          onChange={(e) => onChange()}
        />
        <span>
          {capitalizeFirstWord(spanValue)}
        </span>
      </label>
    </div>
  );
}

export const Checkbox = checkbox;
