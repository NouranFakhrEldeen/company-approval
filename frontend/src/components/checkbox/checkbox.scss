@import '../../styles/variables.scss';

.checkbox-container {
  background-color: transparent;
  padding: 0;

  label {
    background-color: $light-grey;
    box-shadow: none !important;
    border: 1px solid $border_grey;
    margin-top: 5px;
    color: $dark-grey;
    padding: 10px 10px;
    width: 100%;
  }

  span {
    display: inline-block;
    margin-left: 10px;
    color: $black;
  }
}

.width-important {
  label {
    width: 40% !important;
  }
}
