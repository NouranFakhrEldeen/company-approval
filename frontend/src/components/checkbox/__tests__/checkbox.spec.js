import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
// import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Checkbox } from '../checkbox';

let onChange = jest.fn();

describe('testing checkbox', ()=> {
  afterEach(cleanup);

  test('should checkbox checked', () => {
    const { getByTestId } = render(<Checkbox data-testid='checkbox' onChange={()=> onChange()}/>);
    const linkElement = getByTestId('checkbox');
    fireEvent.click(linkElement);
  });

  //   snapshot
  test('should do snapshot', () => {
    const checkboxComp = renderer.create(
      <Checkbox
        onChange={()=> onChange()}
      />
    ).toJSON();
    expect(checkboxComp).toMatchSnapshot();
  });
});

