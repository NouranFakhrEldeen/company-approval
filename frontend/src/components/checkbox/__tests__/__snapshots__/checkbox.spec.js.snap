// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing checkbox should do snapshot 1`] = `
<div>
  <label>
    <input
      data-testid="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
    <span />
  </label>
</div>
`;
